swap_meet/vendor.py

```diff
@@ -1,2 +1,106 @@
 class Vendor:
-    pass
\ No newline at end of file
+    ''' 
+    Represents a Vendor with an invetory of items. 
+    Attributes: inventory with list of items.
+    Methods: add(item): Adds item to inventory, and returns item.
+            remove(item): Removes and returns item, or False if not found.
+    '''
+    
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+  
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_id(self, vendor_id):
+        ''' Retrieves an item from the inventory by its ID.
+            Returns None if no matching item is found.
+        '''
+        
+        for object in self.inventory:
+            if vendor_id == object.id:
+                return object
+        return None   
+        
+    def swap_items(self, other_vendor, my_item, their_item):
+        '''
+        Swaps items between this Vendor and other Vendor.
+        Args: other_vendor(Vendor): the Vendor to swap with.
+              my_item(Item): the item the other Vendor is giving.
+              their_item(Item): the item the other Vendor is giving.
+        Returns: bool: True if the swap was successful, False otherwise.
+        '''
+
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
```

Comment:
We could also use `get_by_id` to check if an item is in a given inventory and then check to see if that function returns `None` or the item.